distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ServerCli.java

```diff
@@ -153,7 +153,9 @@ private Environment autoConfigureSecurity(
             };
             if (options.has(enrollmentTokenOption) == false && okCode) {
                 // we still want to print the error, just don't fail startup
-                terminal.errorPrintln(e.getMessage());
+                if (e.getMessage() != null) {
```

Comment:
Can you add a test for this in ServerCliTests? The mock autoconfig should be able to replicate the conditions.